core-backend/sandbox/src/funcs.rs

```diff
@@ -710,6 +713,22 @@ where
         })
     }
 
+    /// Infallible `gr_panic` syscall.
+    pub fn panic(ctx: &mut Runtime<E>, args: &[Value]) -> SyscallOutput {
+        sys_trace!(target: "syscall::gear", "panic, args = {}", args_to_str(args));
+
+        let (data_ptr, data_len): (_, u32) = args.iter().read_2()?;
+
+        ctx.run(|ctx| {
+            let read_data = ctx.register_read(data_ptr, data_len);
+            let data = ctx.read(read_data)?;
```

Comment:
are you sure we want to exit by trap instead of panic with empty message in case of failing read?
furthermore add check for 0-len to avoid going to memory (open issue)

Reply:
use runtime buffer also?